lib/value/src/value/serde.rs

```diff
@@ -110,12 +113,18 @@ impl<'de> Deserialize<'de> for Value {
             where
                 E: serde::de::Error,
             {
-                Ok(Value::Bytes(Bytes::copy_from_slice(value.as_bytes())))
+                Ok(DateTime::parse_from_rfc3339(value).ok().map_or_else(
+                    || Value::Bytes(Bytes::copy_from_slice(value.as_bytes())),
+                    |dt| Value::Timestamp(dt.with_timezone(&Utc)),
+                ))
```

Comment:
Nit: I don't think `.ok()` is actually needed here as `Result` also has a `map_or_else` method:
```suggestion
                Ok(DateTime::parse_from_rfc3339(value).map_or_else(
                    |_error| Value::Bytes(Bytes::copy_from_slice(value.as_bytes())),
                    |dt| Value::Timestamp(dt.with_timezone(&Utc)),
                ))
```